server/controllers/interview.js

```diff
@@ -0,0 +1,109 @@
+// connect to the Interview Model
+let Interview = require('../models/interview');
+let User = require('../models/user');
+
+module.exports.createInterview = async (req, res, next) => {
+	const { guest, date, theme, questions, difficult } = req.body;
+	const user = await User.findOne({ email: req.user.user }, 'id');
+
+	let newInterview = Interview({
+		owner: user,
+		guest: guest,
+		date: date,
+		theme: theme,
+		questions: questions,
+		difficult: difficult,
+	});
+
+	Interview.create(newInterview, (err) => {
+		if (err) {
+			console.log(err);
+			res.end(err);
+		} else {
+			res.json({ success: true, msg: 'Successfully Created New Interview' });
+		}
+	});
+};
+
+module.exports.getInterviewList = async (req, res, next) => {
+	const user = await User.findOne({ email: req.user.user }, 'id');
+
+	Interview.find(
+		{ $or: [{ owner: user }, { guest: user }] },
+		(err, InterviewList) => {
+			if (err) {
+				console.error(err);
+				res.send(err);
+			} else {
+				res.json(InterviewList);
+			}
+		}
+	);
+};
+
+module.exports.getSingleInterview = (req, res, next) => {
+	const id = req.params.id;
+
+	Interview.findById(id, (err, singleInterview) => {
+		if (err) {
+			console.log(err);
+			res.send(err);
+		} else {
+			res.json(singleInterview);
+		}
+	});
+};
+
+module.exports.updateInterview = (req, res, next) => {
```

Comment:
(optional) It would be worth adding a check to all of these controllers that the user accessing the route is the owner of the specified interview (or guest if they have permissions too)

Reply:
We talk about that in the sprint planning meeting and we decided that we will keep those routes as they are for now